test/unit/mongo/MongoTripodConfigTest.php

```diff
@@ -1380,6 +1380,78 @@ public function testDataLoadedInConfiguredDataSource()
         }
     }
 
+    public function testTransactionLogAndQueuesAreWrittenToCorrectDBAndCollection()
+    {
+        $storeName = 'tripod_php_testing';
+        $newConfig = MongoTripodConfig::getConfig();
+        $newConfig['transaction_log']['database'] = 'tripod_php_testing_transaction_log';
+        $newConfig['transaction_log']['collection'] = 'transaction_log';
+        $newConfig['queue']['database'] = 'tripod_php_testing_queue';
+        $newConfig['queue']['collection'] = 'queue';
+        MongoTripodConfig::setConfig($newConfig);
+
+        $config = MongoTripodConfig::getInstance();
+
+        // Clear out any old data
+        $tlogDB = $config->getTransactionLogDatabase();
+        $tlogDB->drop();
+        $queueDB = $config->getQueueDatabase();
+        $queueDB->drop();
+
+        // Make sure the dbs do not exist
+        $mongo = new MongoClient(MongoTripodConfig::getInstance()->getConnStr('tripod_php_testing'));
+        $dbInfo = $mongo->listDBs();
+        foreach($dbInfo['databases'] as $db){
+            $this->assertFalse(in_array($db['name'], array($newConfig['transaction_log']['database'], $newConfig['queue']['database'])));
+        }
+
+        // Start adding some data
+        $this->tripod = new MongoTripod('CBD_testing', $storeName, array(OP_ASYNC=>array(OP_VIEWS=>true,OP_TABLES=>false,OP_SEARCH=>false)));
+        $this->loadBaseDataViaTripod();
+
+        $graph = new MongoGraph();
+        $subject = 'http://example.com/' . uniqid();
+        $labeller = new MongoTripodLabeller();
+        $graph->add_resource_triple($subject, RDF_TYPE, $labeller->qname_to_uri('foaf:Person'));
+        $graph->add_literal_triple($subject, FOAF_NAME, "Anne Example");
+        $this->tripod->saveChanges(new ExtendedGraph(), $graph);
+
+        $newGraph = $this->tripod->describeResource($subject);
+        $newGraph->add_literal_triple($subject, $labeller->qname_to_uri('foaf:email'), 'anne@example.com');
+        $this->tripod->saveChanges($graph, $newGraph);
+
+        // Make sure the dbs do now exist
+        $mongo = new MongoClient(MongoTripodConfig::getInstance()->getConnStr('tripod_php_testing'));
```

Comment:
If you want to be sure that it's written to the database that was defined in the config, you're going to have to get that from the config document.  For the tlog, you'd have to do something like:
```php 
$tlogDataSource = $config['data_sources'][$config['transaction_log']['data_source']]
```
then, if $tlogDataSource['type'] == 'mongo', you can use the 'connection' and 'replicaSet' properties to init MongoClient with

Queue would be basically the same:
```php
$queueDataSource = $config['data_sources'][$config['queue']['data_source']]
```